state/cluster/badger/mutator.go

```diff
@@ -120,7 +120,25 @@ func (m *MutableState) Extend(block *cluster.Block) error {
 		checkTxsSpan, _ := m.tracer.StartSpanFromContext(ctx, trace.COLClusterStateMutatorExtendCheckTransactionsValid)
 		defer checkTxsSpan.Finish()
 
+		// a valid collection must reference a valid reference block
+		// NOTE: it is valid for a collection to be expired at this point,
+		// otherwise we would compromise liveness of the cluster.
+		refBlock, err := m.headers.ByBlockID(payload.ReferenceBlockID)
+		if errors.Is(err, storage.ErrNotFound) {
+			return state.NewInvalidExtensionErrorf("unknown reference block (id=%x)", payload.ReferenceBlockID)
```

Comment:
In my opinion, this does not always mean that the block/collection is invalid. In my mind, invalid means that some protocol rule has been violated and the proposer of this collection should be slashed. 
However, this error could also occur if this collector is behind on following the main consensus. This is a very different error case. It only means that we can't validate correctness of the collection; as opposed to concluding the collection is invalid.